x-pack/plugins/osquery/server/routes/privileges_check/privileges_check_route.ts

```diff
@@ -22,21 +21,29 @@ export const privilegesCheckRoute = (router: IRouter, osqueryContext: OsqueryApp
     async (context, request, response) => {
       const esClient = context.core.elasticsearch.client.asCurrentUser;
 
-      const privileges = (
-        await esClient.security.hasPrivileges({
-          body: {
-            index: [
-              {
-                names: [`logs-${OSQUERY_INTEGRATION_NAME}.result*`],
-                privileges: ['read'],
-              },
-            ],
-          },
-        })
-      ).body;
+      if (osqueryContext.security.authz.mode.useRbacForRequest(request)) {
+        const privileges = (
+          await esClient.security.hasPrivileges({
+            body: {
+              index: [
+                {
+                  names: [`logs-${OSQUERY_INTEGRATION_NAME}.result*`],
+                  privileges: ['read'],
+                },
+              ],
+            },
+          })
+        ).body;
+
+        return response.ok({
+          body: privileges,
+        });
+      }
 
       return response.ok({
-        body: privileges,
+        body: {
+          has_all_requested: true,
+        },
       });
```

Comment:
This is the important part -- handling the privilege check appropriately if ES does not have Security enabled.

I'm a bit torn as this API is supposed to return the ES privileges response. I understand the easiest approach is what you have here, to just allow the consumer to continue functioning as if the authorization check went all the way through and came back green.

I checked and it appears there is only one consumer of this route, and it just checks if `has_all_requested` is true.
So perhaps the better long-term solution for this would be to change this API to return a boolean, rather than returning the whole ES response. 
WDYT?